src/components/FormStep.js

```diff
@@ -98,14 +98,20 @@ const doLogicCheck = async (stepUrl, data, invalidKeys=[], signal) => {
   const url = `${stepUrl}/_check_logic`;
   // filter out the invalid keys so we only send valid (client-side) input data to the
   // backend to evaluate logic.
+  let dataForLogicCheck = cloneDeep(data);
   if (invalidKeys.length) {
-    data = omit(data, invalidKeys);
+    dataForLogicCheck = omit(dataForLogicCheck, invalidKeys);
   }
```

Comment:
as far as I can see from the docs, `_.omit` should not mutate the first argument, so I think we can avoid the `cloneDeep`?

```js
const dataForLogicCheck = invalidKeys.length ? omit(data, invalidKeys) : data;
```